docs/DafnyRef/Statements.md

```diff
@@ -448,12 +485,12 @@ A failure-compatible type with an `Extract` member is called _value-carrying_.
 To use this form of update,
 
  * if the RHS of the update-with-failure statement is a method call, the first out-parameter of the callee must be failure-compatible
- * if instead the RHS of the update-with-failure statement is one or more expressions, the first of these expressions must be a value with a failure-compatible type
+ * if instead, the RHS of the update-with-failure statement is one or more expressions, the first of these expressions must be a value with a failure-compatible type
  * the caller must have a first out-parameter whose type matches the output of `PropagateFailure` applied to the first output of the callee, unless an
 `expect`, `assume`, or `assert` keyword is used after `:-` (cf. [Section 20.7.7](#sec-failure-return-keyword)).
  * if the failure-compatible type of the RHS does not have an `Extract` member,
 then the LHS of the `:-` statement has one less expression than the RHS
-(or than the number of out-parameters from the method call)
+(or than the number of out-parameters from the method call), the value of the first out-parameter or expression being dropped
```

Comment:
Hem, how is the value of the first out-parameter is being dropped? I though it was taken and just extracted as a boolean.

Reply:
This is the case of a non-value-carrying failure-compatible type, like Result<T>. There is no value to extract so nothing to assign.

Reply:
Still not clear to me in this context. Can you provide an example?

Reply:
There are examples in the next two subsections in the text. Here is another one:
```
module {:options "--function-syntax:4"} M {
  datatype Outcome<E> = Pass | Fail(error: E)
  {
    predicate IsFailure() {
      Fail?
    }
    function PropagateFailure(): Outcome<E>
      requires Fail?
    {
      Fail(this.error)
    }
    // Note: no Extract method
  }

  method m() returns (r: Outcome<string>, t: int) {
    return Fail("bad"), 42;
  }

  method mm() returns  (r: Outcome<string>) {
    var t, tt := m();
    tt :- m();
  }
}
```